openedx/core/djangoapps/content_libraries/api/libraries.py

```diff
@@ -692,3 +689,35 @@ def revert_changes(library_key: LibraryLocatorV2, user_id: int | None = None) ->
 
     # Call the event handlers as needed.
     tasks.wait_for_post_revert_events(draft_change_log, library_key)
+
+
+def get_backup_task_status(
+    user_id: int,
+    task_id: str
+) -> dict | None:
+    """
+    Get the status of a library backup task.
+
+    Returns a dictionary with the following keys:
+        - state: One of "Pending", "Exporting", "Succeeded", "Failed"
+        - url: If state is "Succeeded", the URL where the exported .zip file can be downloaded. Otherwise, None.
+    If no task is found, returns None.
+    """
+    task_status = None
+    if task_id:
+        # Get the specific task by ID
+        try:
+            task_status = UserTaskStatus.objects.get(task_id=task_id, user_id=user_id)
+        except UserTaskStatus.DoesNotExist:
+            return None
+
+    if not task_status:
+        return None
```

Comment:
Since task_id is now required, we don't need all this any longer. We can do something more like:


```suggestion
    try:
        task_status = UserTaskStatus.objects.get(task_id=task_id, user_id=user_id)
    except UserTaskStatus.DoesNotExist:
        return None
```